[Automation] javac compile fails for com.ongres.stringprep:saslprep:2.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean compileTestJava -Pcoordinates="com.ongres.stringprep:saslprep:1.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25310286762/job/74195599837
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
   assertThat(SaslPrep.saslPrep(fullwidthUserWithDigits, true)).isEqualTo("User123");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:40: error: cannot find symbol
        assertThat(SaslPrep.saslPrep(ligatureAndRomanNumeral, true)).isEqualTo("fi-IX");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:45: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("first\u00A0second", true)).isEqualTo("first second");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:46: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("first\u2007second", true)).isEqualTo("first second");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:47: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("first\u202Fsecond", true)).isEqualTo("first second");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:52: error: cannot find symbol
        assertThat(SaslPrep.saslPrep(" leading  inner  trailing ", true))
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:61: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("a" + cjkExtensionB + "\u00AD" + deseretCapitalLetterLongI + "b", true))
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:82: error: cannot find symbol
        assertThat(SaslPrep.saslPrep(stringWithUnassignedCodePoint, false)).isEqualTo(stringWithUnassignedCodePoint);
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:83: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep(stringWithUnassignedCodePoint, true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:91: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("\u0627\u0661\u0628", true)).isEqualTo("\u0627\u0661\u0628");
                   ^
  symbol:   variable SaslPrep

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/build/reports/problems/problems-report.html
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:92: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("\u05D0\u05D1", true)).isEqualTo("\u05D0\u05D1");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:94: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627A\u0628", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:98: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("\u06271", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:102: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("1\u0627", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:109: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("\u0627\uFF11\u0628", true)).isEqualTo("\u06271\u0628");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:111: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627\uFF21\u0628", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:118: error: cannot find symbol
        assertThatNullPointerException().isThrownBy(() -> SaslPrep.saslPrep(null, true));
                                                          ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:122: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep(value, true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
26 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:13: error: cannot find symbol
  import com.ongres.saslprep.SaslPrep;
                            ^
    symbol:   class SaslPrep
    location: package com.ongres.saslprep
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:19: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("I\u00ADX", true)).isEqualTo("IX");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:20: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("user", true)).isEqualTo("user");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:21: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("USER", true)).isEqualTo("USER");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:22: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u00AA", true)).isEqualTo("a");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:23: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u2168", true)).isEqualTo("IX");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:24: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\uD840\uDC00", true)).isEqualTo("\uD840\uDC00");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:29: error: cannot find symbol
          String prepared = SaslPrep.saslPrep("pass\u00AD\u034Fword\u200B\uFE00", true);
                            ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:39: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(fullwidthUserWithDigits, true)).isEqualTo("User123");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:40: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(ligatureAndRomanNumeral, true)).isEqualTo("fi-IX");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:45: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("first\u00A0second", true)).isEqualTo("first second");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:46: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("first\u2007second", true)).isEqualTo("first second");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:47: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("first\u202Fsecond", true)).isEqualTo("first second");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:52: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(" leading  inner  trailing ", true))
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:61: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("a" + cjkExtensionB + "\u00AD" + deseretCapitalLetterLongI + "b", true))
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:82: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(stringWithUnassignedCodePoint, false)).isEqualTo(stringWithUnassignedCodePoint);
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:83: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep(stringWithUnassignedCodePoint, true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:91: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u0627\u0661\u0628", true)).isEqualTo("\u0627\u0661\u0628");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:92: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u05D0\u05D1", true)).isEqualTo("\u05D0\u05D1");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:94: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627A\u0628", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:98: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("\u06271", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:102: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("1\u0627", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:109: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u0627\uFF11\u0628", true)).isEqualTo("\u06271\u0628");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:111: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627\uFF21\u0628", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:118: error: cannot find symbol
          assertThatNullPointerException().isThrownBy(() -> SaslPrep.saslPrep(null, true));
                                                            ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:122: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep(value, true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  26 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0 at 2026-05-04T09:03:46Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 2.0 failed with exit code 1.
FAILED[javac compile][2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean compileTestJava -Pcoordinates="com.ongres.stringprep:saslprep:1.1"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean compileTestJava -Pcoordinates="com.ongres.stringprep:saslprep:1.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25310286762/job/74195599818
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
   assertThat(SaslPrep.saslPrep(fullwidthUserWithDigits, true)).isEqualTo("User123");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:40: error: cannot find symbol
        assertThat(SaslPrep.saslPrep(ligatureAndRomanNumeral, true)).isEqualTo("fi-IX");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:45: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("first\u00A0second", true)).isEqualTo("first second");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:46: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("first\u2007second", true)).isEqualTo("first second");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:47: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("first\u202Fsecond", true)).isEqualTo("first second");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:52: error: cannot find symbol
        assertThat(SaslPrep.saslPrep(" leading  inner  trailing ", true))
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:61: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("a" + cjkExtensionB + "\u00AD" + deseretCapitalLetterLongI + "b", true))
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:82: error: cannot find symbol
        assertThat(SaslPrep.saslPrep(stringWithUnassignedCodePoint, false)).isEqualTo(stringWithUnassignedCodePoint);
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:83: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep(stringWithUnassignedCodePoint, true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:91: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("\u0627\u0661\u0628", true)).isEqualTo("\u0627\u0661\u0628");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:92: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("\u05D0\u05D1", true)).isEqualTo("\u05D0\u05D1");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:94: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627A\u0628", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:98: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("\u06271", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:102: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("1\u0627", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:109: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("\u0627\uFF11\u0628", true)).isEqualTo("\u06271\u0628");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:111: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627\uFF21\u0628", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:118: error: cannot find symbol
        assertThatNullPointerException().isThrownBy(() -> SaslPrep.saslPrep(null, true));
                                                          ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:122: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep(value, true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
26 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:13: error: cannot find symbol
  import com.ongres.saslprep.SaslPrep;
                            ^
    symbol:   class SaslPrep
    location: package com.ongres.saslprep
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:19: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("I\u00ADX", true)).isEqualTo("IX");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:20: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("user", true)).isEqualTo("user");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:21: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("USER", true)).isEqualTo("USER");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:22: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u00AA", true)).isEqualTo("a");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:23: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u2168", true)).isEqualTo("IX");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:24: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\uD840\uDC00", true)).isEqualTo("\uD840\uDC00");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:29: error: cannot find symbol
          String prepared = SaslPrep.saslPrep("pass\u00AD\u034Fword\u200B\uFE00", true);
                            ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:39: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(fullwidthUserWithDigits, true)).isEqualTo("User123");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:40: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(ligatureAndRomanNumeral, true)).isEqualTo("fi-IX");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:45: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("first\u00A0second", true)).isEqualTo("first second");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:46: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("first\u2007second", true)).isEqualTo("first second");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:47: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("first\u202Fsecond", true)).isEqualTo("first second");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:52: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(" leading  inner  trailing ", true))
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:61: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("a" + cjkExtensionB + "\u00AD" + deseretCapitalLetterLongI + "b", true))
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:82: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(stringWithUnassignedCodePoint, false)).isEqualTo(stringWithUnassignedCodePoint);
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:83: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep(stringWithUnassignedCodePoint, true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:91: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u0627\u0661\u0628", true)).isEqualTo("\u0627\u0661\u0628");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:92: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u05D0\u05D1", true)).isEqualTo("\u05D0\u05D1");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/build/reports/problems/problems-report.html
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:94: error: cannot find symbol

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date
          assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627A\u0628", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:98: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("\u06271", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:102: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("1\u0627", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:109: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u0627\uFF11\u0628", true)).isEqualTo("\u06271\u0628");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:111: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627\uFF21\u0628", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:118: error: cannot find symbol
          assertThatNullPointerException().isThrownBy(() -> SaslPrep.saslPrep(null, true));
                                                            ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:122: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep(value, true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  26 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0 at 2026-05-04T09:03:50Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 2.0 failed with exit code 1.
FAILED[javac compile][2.0][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="2.0" ./gradlew clean compileTestJava -Pcoordinates="com.ongres.stringprep:saslprep:1.1"]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.0" ./gradlew clean compileTestJava -Pcoordinates="com.ongres.stringprep:saslprep:1.1"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25310286762/job/74195599774
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
  assertThat(SaslPrep.saslPrep(fullwidthUserWithDigits, true)).isEqualTo("User123");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:40: error: cannot find symbol
        assertThat(SaslPrep.saslPrep(ligatureAndRomanNumeral, true)).isEqualTo("fi-IX");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:45: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("first\u00A0second", true)).isEqualTo("first second");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:46: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("first\u2007second", true)).isEqualTo("first second");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:47: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("first\u202Fsecond", true)).isEqualTo("first second");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:52: error: cannot find symbol
        assertThat(SaslPrep.saslPrep(" leading  inner  trailing ", true))
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:61: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("a" + cjkExtensionB + "\u00AD" + deseretCapitalLetterLongI + "b", true))
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:82: error: cannot find symbol
        assertThat(SaslPrep.saslPrep(stringWithUnassignedCodePoint, false)).isEqualTo(stringWithUnassignedCodePoint);
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:83: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep(stringWithUnassignedCodePoint, true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:91: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("\u0627\u0661\u0628", true)).isEqualTo("\u0627\u0661\u0628");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:92: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("\u05D0\u05D1", true)).isEqualTo("\u05D0\u05D1");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:94: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627A\u0628", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:98: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("\u06271", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:102: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("1\u0627", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:109: error: cannot find symbol
        assertThat(SaslPrep.saslPrep("\u0627\uFF11\u0628", true)).isEqualTo("\u06271\u0628");
                   ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:111: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627\uFF21\u0628", true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:118: error: cannot find symbol
        assertThatNullPointerException().isThrownBy(() -> SaslPrep.saslPrep(null, true));
                                                          ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:122: error: cannot find symbol
        assertThatThrownBy(() -> SaslPrep.saslPrep(value, true))
                                 ^
  symbol:   variable SaslPrep
  location: class SaslprepTest
26 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:13: error: cannot find symbol
  import com.ongres.saslprep.SaslPrep;
                            ^
    symbol:   class SaslPrep
    location: package com.ongres.saslprep
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:19: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("I\u00ADX", true)).isEqualTo("IX");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:20: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("user", true)).isEqualTo("user");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:21: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("USER", true)).isEqualTo("USER");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:22: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u00AA", true)).isEqualTo("a");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:23: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u2168", true)).isEqualTo("IX");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:24: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\uD840\uDC00", true)).isEqualTo("\uD840\uDC00");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:29: error: cannot find symbol
          String prepared = SaslPrep.saslPrep("pass\u00AD\u034Fword\u200B\uFE00", true);
                            ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:39: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(fullwidthUserWithDigits, true)).isEqualTo("User123");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:40: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(ligatureAndRomanNumeral, true)).isEqualTo("fi-IX");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:45: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("first\u00A0second", true)).isEqualTo("first second");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:46: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("first\u2007second", true)).isEqualTo("first second");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:47: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("first\u202Fsecond", true)).isEqualTo("first second");
                     ^
    symbol:   variable SaslPrep
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/build/reports/problems/problems-report.html

    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:52: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(" leading  inner  trailing ", true))
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:61: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("a" + cjkExtensionB + "\u00AD" + deseretCapitalLetterLongI + "b", true))
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:82: error: cannot find symbol
          assertThat(SaslPrep.saslPrep(stringWithUnassignedCodePoint, false)).isEqualTo(stringWithUnassignedCodePoint);
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:83: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep(stringWithUnassignedCodePoint, true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:91: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u0627\u0661\u0628", true)).isEqualTo("\u0627\u0661\u0628");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:92: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u05D0\u05D1", true)).isEqualTo("\u05D0\u05D1");
                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:94: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627A\u0628", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:98: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("\u06271", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:102: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("1\u0627", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:109: error: cannot find symbol
          assertThat(SaslPrep.saslPrep("\u0627\uFF11\u0628", true)).isEqualTo("\u06271\u0628");
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

                     ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:111: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627\uFF21\u0628", true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:118: error: cannot find symbol
          assertThatNullPointerException().isThrownBy(() -> SaslPrep.saslPrep(null, true));
                                                            ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/com.ongres.stringprep/saslprep/1.1/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java:122: error: cannot find symbol
          assertThatThrownBy(() -> SaslPrep.saslPrep(value, true))
                                   ^
    symbol:   variable SaslPrep
    location: class SaslprepTest
  26 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.0 at 2026-05-04T09:04:08Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 2.0 failed with exit code 1.
FAILED[javac compile][2.0][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="2.0" ./gradlew clean compileTestJava -Pcoordinates="com.ongres.stringprep:saslprep:1.1"]
```

--- a/metadata/com.ongres.stringprep/saslprep/2.0/reachability-metadata.json
+++ b/metadata/com.ongres.stringprep/saslprep/2.0/reachability-metadata.json
@@ -1,0 +1,11 @@
+{
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "com.ongres.saslprep.SASLprep"
+      },
+      "module" : "java.base",
+      "glob" : "jdk/internal/icu/impl/data/icudt76b/nfkc.nrm"
+    }
+  ]
+}

--- a/metadata/com.ongres.stringprep/saslprep/index.json
+++ b/metadata/com.ongres.stringprep/saslprep/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "2.0",
+    "tested-versions" : [
+      "2.0"
+    ],
+    "allowed-packages" : [
+      "com.ongres.saslprep"
+    ]
+  },
+  {
     "metadata-version" : "1.1",
     "source-code-url" : "https://repo.maven.apache.org/maven2/com/ongres/stringprep/saslprep/$version$/saslprep-$version$-sources.jar",
     "repository-url" : "https://github.com/ongres/stringprep",

--- a/metadata/com.ongres.stringprep/saslprep/index.json
+++ b/metadata/com.ongres.stringprep/saslprep/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "2.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/com/ongres/stringprep/saslprep/$version$/saslprep-$version$-sources.jar",
+    "repository-url" : "https://github.com/ongres/stringprep",
+    "test-code-url" : "https://github.com/ongres/stringprep/tree/$version$/saslprep/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/com/ongres/stringprep/saslprep/$version$/saslprep-$version$-javadoc.jar",
+    "description" : "SASLprep implements the RFC 4013 profile of StringPrep for preparing user names and passwords in Java applications. It maps, normalizes, validates, and checks bidirectional text rules for strings before they are used in SASL authentication flows.",
     "tested-versions" : [
       "2.0"
     ],

--- a/stats/com.ongres.stringprep/saslprep/2.0/execution-metrics.json
+++ b/stats/com.ongres.stringprep/saslprep/2.0/execution-metrics.json
@@ -1,0 +1,88 @@
+{
+  "fix_javac_fail:2026-06-06": {
+    "timestamp": "2026-06-06T06:16:06.859019Z",
+    "library": "com.ongres.stringprep:saslprep:2.0",
+    "starting_commit": "0cad58abf642dd87ae2cd32d766081d29bf8107c",
+    "ending_commit": "40bb601603166640197c21b74609650b1d30f047",
+    "previous_library": "com.ongres.stringprep:saslprep:1.1",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.0",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 85,
+          "missed": 0,
+          "ratio": 1.0,
+          "total": 85
+        },
+        "line": {
+          "covered": 7,
+          "missed": 0,
+          "ratio": 1.0,
+          "total": 7
+        },
+        "method": {
+          "covered": 5,
+          "missed": 0,
+          "ratio": 1.0,
+          "total": 5
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "1.1",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 183,
+          "missed": 3,
+          "ratio": 0.983871,
+          "total": 186
+        },
+        "line": {
+          "covered": 44,
+          "missed": 1,
+          "ratio": 0.977778,
+          "total": 45
+        },
+        "method": {
+          "covered": 1,
+          "missed": 1,
+          "ratio": 0.5,
+          "total": 2
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 35272,
+      "cached_input_tokens_used": 211456,
+      "output_tokens_used": 3103,
+      "iterations": 2,
+      "cost_usd": 0.1988,
+      "tested_library_loc": 7,
+      "metadata_entries": 2,
+      "test_only_metadata_entries": 2,
+      "previous_library_metadata_entries": 0,
+      "code_coverage_percent": 100.0,
+      "previous_library_coverage_percent": 97.78
+    },
+    "artifacts": {
+      "test_file": "tests/src/com.ongres.stringprep/saslprep/2.0/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java",
+      "metadata_file": "metadata/com.ongres.stringprep/saslprep/2.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/com.ongres.stringprep/saslprep/2.0/stats.json
+++ b/stats/com.ongres.stringprep/saslprep/2.0/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "2.0",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 85,
+        "missed" : 0,
+        "ratio" : 1.0,
+        "total" : 85
+      },
+      "line" : {
+        "covered" : 7,
+        "missed" : 0,
+        "ratio" : 1.0,
+        "total" : 7
+      },
+      "method" : {
+        "covered" : 5,
+        "missed" : 0,
+        "ratio" : 1.0,
+        "total" : 5
+      }
+    }
+  } ]
+}

--- a/tests/src/com.ongres.stringprep/saslprep/2.0/.gitignore
+++ b/tests/src/com.ongres.stringprep/saslprep/2.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/com.ongres.stringprep/saslprep/2.0/build.gradle
+++ b/tests/src/com.ongres.stringprep/saslprep/2.0/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "com.ongres.stringprep:saslprep:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/com.ongres.stringprep/saslprep/2.0/gradle.properties
+++ b/tests/src/com.ongres.stringprep/saslprep/2.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = com.ongres.stringprep:saslprep:2.0
+library.version = 2.0
+metadata.dir = com.ongres.stringprep/saslprep/2.0/

--- a/tests/src/com.ongres.stringprep/saslprep/2.0/settings.gradle
+++ b/tests/src/com.ongres.stringprep/saslprep/2.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'com.ongres.stringprep.saslprep_tests'

--- a/tests/src/com.ongres.stringprep/saslprep/2.0/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java
+++ b/tests/src/com.ongres.stringprep/saslprep/2.0/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java
@@ -1,0 +1,127 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_ongres_stringprep.saslprep;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNullPointerException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.ongres.saslprep.SaslPrep;
+import org.junit.jupiter.api.Test;
+
+public class SaslprepTest {
+    @Test
+    void preparesRfc4013Examples() {
+        assertThat(SaslPrep.saslPrep("I\u00ADX", true)).isEqualTo("IX");
+        assertThat(SaslPrep.saslPrep("user", true)).isEqualTo("user");
+        assertThat(SaslPrep.saslPrep("USER", true)).isEqualTo("USER");
+        assertThat(SaslPrep.saslPrep("\u00AA", true)).isEqualTo("a");
+        assertThat(SaslPrep.saslPrep("\u2168", true)).isEqualTo("IX");
+        assertThat(SaslPrep.saslPrep("\uD840\uDC00", true)).isEqualTo("\uD840\uDC00");
+    }
+
+    @Test
+    void removesCharactersCommonlyMappedToNothingBeforeValidation() {
+        String prepared = SaslPrep.saslPrep("pass\u00AD\u034Fword\u200B\uFE00", true);
+
+        assertThat(prepared).isEqualTo("password");
+    }
+
+    @Test
+    void appliesNfkcCompatibilityNormalizationWithoutCaseFolding() {
+        String fullwidthUserWithDigits = "\uFF35\uFF53\uFF45\uFF52\uFF11\uFF12\uFF13";
+        String ligatureAndRomanNumeral = "\uFB01-\u2168";
+
+        assertThat(SaslPrep.saslPrep(fullwidthUserWithDigits, true)).isEqualTo("User123");
+        assertThat(SaslPrep.saslPrep(ligatureAndRomanNumeral, true)).isEqualTo("fi-IX");
+    }
+
+    @Test
+    void mapsNonAsciiSpaceCharactersToAsciiSpaceDuringPreparation() {
+        assertThat(SaslPrep.saslPrep("first\u00A0second", true)).isEqualTo("first second");
+        assertThat(SaslPrep.saslPrep("first\u2007second", true)).isEqualTo("first second");
+        assertThat(SaslPrep.saslPrep("first\u202Fsecond", true)).isEqualTo("first second");
+    }
+
+    @Test
+    void preservesAsciiSpacesWithoutTrimmingOrCollapsing() {
+        assertThat(SaslPrep.saslPrep(" leading  inner  trailing ", true))
+                .isEqualTo(" leading  inner  trailing ");
+    }
+
+    @Test
+    void handlesSupplementaryCodePointsWithoutSplittingSurrogatePairs() {
+        String cjkExtensionB = "\uD840\uDC00";
+        String deseretCapitalLetterLongI = "\uD801\uDC00";
+
+        assertThat(SaslPrep.saslPrep("a" + cjkExtensionB + "\u00AD" + deseretCapitalLetterLongI + "b", true))
+                .isEqualTo("a" + cjkExtensionB + deseretCapitalLetterLongI + "b");
+    }
+
+    @Test
+    void rejectsProhibitedCharactersFromEachSaslprepCategory() {
+        assertProhibited("ASCII control", "ok\u0007");
+        assertProhibited("non-ASCII control", "ok\u0080");
+        assertProhibited("private use", "ok\uE000");
+        assertProhibited("non-character code point", "ok\uFDD0");
+        assertProhibited("surrogate code point", "ok\uD800");
+        assertProhibited("plain text inappropriate", "ok\uFFF9");
+        assertProhibited("canonical representation inappropriate", "ok\u2FF0");
+        assertProhibited("display property changing", "ok\u200E");
+        assertProhibited("tagging", "ok\uDB40\uDC01");
+    }
+
+    @Test
+    void rejectsUnassignedCodePointsOnlyForStoredStrings() {
+        String stringWithUnassignedCodePoint = "user\u0221";
+
+        assertThat(SaslPrep.saslPrep(stringWithUnassignedCodePoint, false)).isEqualTo(stringWithUnassignedCodePoint);
+        assertThatThrownBy(() -> SaslPrep.saslPrep(stringWithUnassignedCodePoint, true))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Prohibited character")
+                .hasMessageContaining("\u0221");
+    }
+
+    @Test
+    void enforcesBidirectionalRuleForRandAlCatCharacters() {
+        assertThat(SaslPrep.saslPrep("\u0627\u0661\u0628", true)).isEqualTo("\u0627\u0661\u0628");
+        assertThat(SaslPrep.saslPrep("\u05D0\u05D1", true)).isEqualTo("\u05D0\u05D1");
+
+        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627A\u0628", true))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("RandALCat and LCat");
+
+        assertThatThrownBy(() -> SaslPrep.saslPrep("\u06271", true))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("RandALCat character is not the first and the last characters");
+
+        assertThatThrownBy(() -> SaslPrep.saslPrep("1\u0627", true))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("RandALCat character is not the first and the last characters");
+    }
+
+    @Test
+    void appliesBidirectionalRuleAfterCompatibilityNormalization() {
+        assertThat(SaslPrep.saslPrep("\u0627\uFF11\u0628", true)).isEqualTo("\u06271\u0628");
+
+        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627\uFF21\u0628", true))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("RandALCat and LCat");
+    }
+
+    @Test
+    void rejectsNullInput() {
+        assertThatNullPointerException().isThrownBy(() -> SaslPrep.saslPrep(null, true));
+    }
+
+    private static void assertProhibited(String description, String value) {
+        assertThatThrownBy(() -> SaslPrep.saslPrep(value, true))
+                .as(description)
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("Prohibited character");
+    }
+}

--- a/tests/src/com.ongres.stringprep/saslprep/2.0/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java
+++ b/tests/src/com.ongres.stringprep/saslprep/2.0/src/test/java/com_ongres_stringprep/saslprep/SaslprepTest.java
@@ -10,23 +10,25 @@ import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatNullPointerException;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
-import com.ongres.saslprep.SaslPrep;
+import com.ongres.saslprep.SASLprep;
 import org.junit.jupiter.api.Test;
 
 public class SaslprepTest {
+    private static final SASLprep SASL_PREP = new SASLprep();
+
     @Test
     void preparesRfc4013Examples() {
-        assertThat(SaslPrep.saslPrep("I\u00ADX", true)).isEqualTo("IX");
-        assertThat(SaslPrep.saslPrep("user", true)).isEqualTo("user");
-        assertThat(SaslPrep.saslPrep("USER", true)).isEqualTo("USER");
-        assertThat(SaslPrep.saslPrep("\u00AA", true)).isEqualTo("a");
-        assertThat(SaslPrep.saslPrep("\u2168", true)).isEqualTo("IX");
-        assertThat(SaslPrep.saslPrep("\uD840\uDC00", true)).isEqualTo("\uD840\uDC00");
+        assertThat(prepare("I\u00ADX", true)).isEqualTo("IX");
+        assertThat(prepare("user", true)).isEqualTo("user");
+        assertThat(prepare("USER", true)).isEqualTo("USER");
+        assertThat(prepare("\u00AA", true)).isEqualTo("a");
+        assertThat(prepare("\u2168", true)).isEqualTo("IX");
+        assertThat(prepare("\uD840\uDC00", true)).isEqualTo("\uD840\uDC00");
     }
 
     @Test
     void removesCharactersCommonlyMappedToNothingBeforeValidation() {
-        String prepared = SaslPrep.saslPrep("pass\u00AD\u034Fword\u200B\uFE00", true);
+        String prepared = prepare("pass\u00AD\u034Fword\u200B\uFE00", true);
 
         assertThat(prepared).isEqualTo("password");
     }
@@ -36,20 +38,20 @@ public class SaslprepTest {
         String fullwidthUserWithDigits = "\uFF35\uFF53\uFF45\uFF52\uFF11\uFF12\uFF13";
         String ligatureAndRomanNumeral = "\uFB01-\u2168";
 
-        assertThat(SaslPrep.saslPrep(fullwidthUserWithDigits, true)).isEqualTo("User123");
-        assertThat(SaslPrep.saslPrep(ligatureAndRomanNumeral, true)).isEqualTo("fi-IX");
+        assertThat(prepare(fullwidthUserWithDigits, true)).isEqualTo("User123");
+        assertThat(prepare(ligatureAndRomanNumeral, true)).isEqualTo("fi-IX");
     }
 
     @Test
     void mapsNonAsciiSpaceCharactersToAsciiSpaceDuringPreparation() {
-        assertThat(SaslPrep.saslPrep("first\u00A0second", true)).isEqualTo("first second");
-        assertThat(SaslPrep.saslPrep("first\u2007second", true)).isEqualTo("first second");
-        assertThat(SaslPrep.saslPrep("first\u202Fsecond", true)).isEqualTo("first second");
+        assertThat(prepare("first\u00A0second", true)).isEqualTo("first second");
+        assertThat(prepare("first\u2007second", true)).isEqualTo("first second");
+        assertThat(prepare("first\u202Fsecond", true)).isEqualTo("first second");
     }
 
     @Test
     void preservesAsciiSpacesWithoutTrimmingOrCollapsing() {
-        assertThat(SaslPrep.saslPrep(" leading  inner  trailing ", true))
+        assertThat(prepare(" leading  inner  trailing ", true))
                 .isEqualTo(" leading  inner  trailing ");
     }
 
@@ -58,7 +60,7 @@ public class SaslprepTest {
         String cjkExtensionB = "\uD840\uDC00";
         String deseretCapitalLetterLongI = "\uD801\uDC00";
 
-        assertThat(SaslPrep.saslPrep("a" + cjkExtensionB + "\u00AD" + deseretCapitalLetterLongI + "b", true))
+        assertThat(prepare("a" + cjkExtensionB + "\u00AD" + deseretCapitalLetterLongI + "b", true))
                 .isEqualTo("a" + cjkExtensionB + deseretCapitalLetterLongI + "b");
     }
 
@@ -79,49 +81,56 @@ public class SaslprepTest {
     void rejectsUnassignedCodePointsOnlyForStoredStrings() {
         String stringWithUnassignedCodePoint = "user\u0221";
 
-        assertThat(SaslPrep.saslPrep(stringWithUnassignedCodePoint, false)).isEqualTo(stringWithUnassignedCodePoint);
-        assertThatThrownBy(() -> SaslPrep.saslPrep(stringWithUnassignedCodePoint, true))
+        assertThat(prepare(stringWithUnassignedCodePoint, false)).isEqualTo(stringWithUnassignedCodePoint);
+        assertThatThrownBy(() -> prepare(stringWithUnassignedCodePoint, true))
                 .isInstanceOf(IllegalArgumentException.class)
-                .hasMessageContaining("Prohibited character")
-                .hasMessageContaining("\u0221");
+                .hasMessageContaining("Unassigned code point")
+                .hasMessageContaining("0x0221");
     }
 
     @Test
     void enforcesBidirectionalRuleForRandAlCatCharacters() {
-        assertThat(SaslPrep.saslPrep("\u0627\u0661\u0628", true)).isEqualTo("\u0627\u0661\u0628");
-        assertThat(SaslPrep.saslPrep("\u05D0\u05D1", true)).isEqualTo("\u05D0\u05D1");
+        assertThat(prepare("\u0627\u0661\u0628", true)).isEqualTo("\u0627\u0661\u0628");
+        assertThat(prepare("\u05D0\u05D1", true)).isEqualTo("\u05D0\u05D1");
 
-        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627A\u0628", true))
+        assertThatThrownBy(() -> prepare("\u0627A\u0628", true))
                 .isInstanceOf(IllegalArgumentException.class)
                 .hasMessageContaining("RandALCat and LCat");
 
-        assertThatThrownBy(() -> SaslPrep.saslPrep("\u06271", true))
+        assertThatThrownBy(() -> prepare("\u06271", true))
                 .isInstanceOf(IllegalArgumentException.class)
-                .hasMessageContaining("RandALCat character is not the first and the last characters");
+                .hasMessageContaining("RandALCat character is not the first and the last character");
 
-        assertThatThrownBy(() -> SaslPrep.saslPrep("1\u0627", true))
+        assertThatThrownBy(() -> prepare("1\u0627", true))
                 .isInstanceOf(IllegalArgumentException.class)
-                .hasMessageContaining("RandALCat character is not the first and the last characters");
+                .hasMessageContaining("RandALCat character is not the first and the last character");
     }
 
     @Test
     void appliesBidirectionalRuleAfterCompatibilityNormalization() {
-        assertThat(SaslPrep.saslPrep("\u0627\uFF11\u0628", true)).isEqualTo("\u06271\u0628");
+        assertThat(prepare("\u0627\uFF11\u0628", true)).isEqualTo("\u06271\u0628");
 
-        assertThatThrownBy(() -> SaslPrep.saslPrep("\u0627\uFF21\u0628", true))
+        assertThatThrownBy(() -> prepare("\u0627\uFF21\u0628", true))
                 .isInstanceOf(IllegalArgumentException.class)
                 .hasMessageContaining("RandALCat and LCat");
     }
 
     @Test
     void rejectsNullInput() {
-        assertThatNullPointerException().isThrownBy(() -> SaslPrep.saslPrep(null, true));
+        assertThatNullPointerException().isThrownBy(() -> prepare(null, true));
     }
 
     private static void assertProhibited(String description, String value) {
-        assertThatThrownBy(() -> SaslPrep.saslPrep(value, true))
+        assertThatThrownBy(() -> prepare(value, true))
                 .as(description)
                 .isInstanceOf(IllegalArgumentException.class)
-                .hasMessageContaining("Prohibited character");
+                .hasMessageContaining("Prohibited");
+    }
+
+    private static String prepare(String value, boolean stored) {
+        if (stored) {
+            return SASL_PREP.prepareStored(value);
+        }
+        return SASL_PREP.prepareQuery(value);
     }
 }

--- a/tests/src/com.ongres.stringprep/saslprep/2.0/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/com.ongres.stringprep/saslprep/2.0/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,16 @@
+{
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "com_ongres_stringprep.saslprep.SaslprepTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.configuration.Configuration"
+    },
+    {
+      "condition" : {
+        "typeReached" : "com_ongres_stringprep.saslprep.SaslprepTest"
+      },
+      "glob" : "META-INF/services/org.assertj.core.presentation.Representation"
+    }
+  ]
+}

--- a/tests/src/com.ongres.stringprep/saslprep/2.0/user-code-filter.json
+++ b/tests/src/com.ongres.stringprep/saslprep/2.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "com.ongres.saslprep.**"
+    },
+    {
+      "includeClasses" : "com_ongres_stringprep.saslprep.**"
+    }
+  ]
+}

